### ISSUE TITLE
`if` after `else` in `try` raises false positive WPS513

### ISSUE THREAD

**Issue by sobolevn** [wemake-services/wemake-python-styleguide#835]
# Bug report

## What's wrong

Code:

```python
        try:
            return_value = ast.literal_eval(node.body)
        except ValueError:
            # The value was not literal, it's ok.
            return
        else:
            if return_value in self._primitive_values:
                self.add_violation(ImplicitPrimitiveViolation(node))
```

Output:

```
329:13   WPS513 Found implicit `elif` condition
  if return_value in self._primitive_values:
```

## How is that should be

It should not raise this violation. It is not a part of `if` / `else` case.

**Comment by lensvol** [comment]
Can I take this?

**Comment by sobolevn** [comment]
@lensvol sure! Do you have any ideas how to implement it? Because, I don't 😞

**Comment by lensvol** [comment]
I do have an idea and it will sound _weird_. Here it is:

The cause of the problem is the assumption that `else` token can only be applied to `if` statement, but `except` and `for` can also be followed by this token. So to prevent this error we should introduce an additional check that `else` is part of the previous `if` statement.

Here it gets tricky. 

To achieve this I propose we use a heuristic that relies on a fact that Python uses indents to distinguish between different code blocks. What we need is to find a closest preceding token with the same level of indentation and see if it is either `if` or `elif`. If it is `except` or `for`, then just carry on.

That's the idea in a nutshell. Can't say for sure that it will work (corner cases etc), but worth exploring in my opinion.

**Comment by sobolevn** [comment]
Awesome, let's do it!
